File: examples/typescript/register-agent.ts
import { KSwitchClient } from "@kswitch/sdk";

const client = KSwitchClient.fromEnv();

const agent = await client.governance.registerAgent({
  display_name: "customer-onboarding-v1",
  record_type: "AGENT",
  risk_tier: "tier_2",
  owning_division: "Retail Banking",
  owning_team: "onboarding-platform",
});

await client.governance.connectMcps(String(agent.id), {
  mcp_ids: ["mcp-kyc", "mcp-customer-data"],
});

console.log(agent.id);

File: examples/python/register_agent.py
from kswitch import KSwitchClient


def main() -> None:
    client = KSwitchClient.from_env()
    agent = client.governance.register_agent(
        display_name="customer-onboarding-v1",
        record_type="AGENT",
        risk_tier="tier_2",
        owning_division="Retail Banking",
        owning_team="onboarding-platform",
    )
    client.governance.connect_mcps(agent["id"], mcp_ids=["mcp-kyc", "mcp-customer-data"])
    print(agent["id"])


if __name__ == "__main__":
    main()

